components/supervisor/pkg/ports/ports.go

```diff
@@ -795,6 +795,22 @@ func (pm *Manager) getPortStatus(port uint32) *api.PortsStatus {
 	return ps
 }
 
+func (pm *Manager) AddInternalPort(port uint32) {
```

Comment:
Should not we just list them as initial internal ports here: https://github.com/gitpod-io/gitpod/blob/53f82fe9bc5cbc03541185d24cc60db351b215b6/components/supervisor/pkg/supervisor/supervisor.go#L208-L211

Reply:
in this case you don't add it here?

Reply:
I don't think we should conflate statically known and dynamic ports. If we know in advance port we should mark it as internal immediately that it does not happen that it pops up first in ports view and then disappear.

With dynamic I agree but I am not sure how to handle them yet, we need some communicate channel from supervisor and desktop IDE startup code then. Ports can change because of underlying IDE was restarted and so on. Let's tackle in another PR and file an issue for now? probably we can handle it after https://github.com/gitpod-io/gitpod/issues/6545

Reply:
I think it is alright, I would change to:
```
desktopIDEPort := 24000
if cfg.DesktopIDE != nil {
	internalPorts = append(internalPorts, uint32(desktopIDEPort))
}
```

`desktopIDEPort` should be passed to desktop IDEs, i.e. hardcoded only in supervisor, instead of hardcoding in each IDE

Reply:
yes